### ISSUE TITLE
PRIORITY: CI

### ISSUE THREAD

**Issue by magicaldave** [DreamWeave-MP/Dreamweave#1]
CI, CI, CI. Manual builds should not be a thing for releases if it can be avoided.

**Comment by magicaldave** [comment]
Linux CI has been functional for a while. Windows builds are now achieved, with only the packaging and some additional libraries needing to be built by cmake for this to be closed.

**Comment by magicaldave** [comment]
Dah, dah, dah, dah, Github.

Slow your roll.

#1  is still active.

Here's what's missing:

- [ ] server-only builds for all platforms
- [ ] macOS builds
- [ ] ARM builds

**Comment by HotaruBlaze** [comment]
> * jobs need to generate releases when pushing to master

Do we really want to do that, that seems like its going to generate too many builds and just confuse people trying to update/install a specific versions, we should probably do targeted releases.

**Comment by YvanCywan** [comment]
> > * jobs need to generate releases when pushing to master
> 
> Do we really want to do that, that seems like its going to generate too many builds and just confuse people trying to update/install a specific versions, we should probably do targeted releases.

This is a fair point, we should tag releases and then use feature flags so that we can patch when we needed.

**Comment by magicaldave** [comment]
These are good callouts; we don't necessarily need to be spamming prerelease builds as we've already got the artifacts. What we really need is to optionally trigger them on branches and create tagged releases from that.

**Comment by YvanCywan** [comment]
don't need to do it on branches. The main branch should always be production ready anyway.

**Comment by HotaruBlaze** [comment]
Building on every merge to master is going to get messy, especially when we usually bulk merge 2-3 at a time

**Comment by YvanCywan** [comment]
We should be building on every push to master anyway. That is literally a requirement to ensure that our changes are integrated properly. But for a proper deployment/release, we need to tag a release to create a release build.

**Comment by HotaruBlaze** [comment]
yeah that was what i saying, that time i said building instead of releasing, I've fixed the typo

**Comment by magicaldave** [comment]
> Releasing on every merge to master is going to get messy, especially when we usually bulk merge 2-3 at a time
> 
> I'd rather just tag master, and for that tag to trigger CI and a release.

Well, we don't *need* to do this anyway as we have access to the build artifacts, but there's no reason it would necessarily create *new* releases every time. The way I've structured the subrepos such as CJson work this way, there's just a single tag they release to instead of, say, one per commit.

**Comment by HotaruBlaze** [comment]
Telling users to access artifacts isn't clean nor a controllable way to have users pick a version, you'd still should be doing tagged releases like tes3mp does.

**Comment by YvanCywan** [comment]
Artifacts should be handled by our CD systems anyway, use of proper build tools integrated with our CD will lead to release builds being packaged with those artifacts. Meaning that users will have to do nothing.

**Comment by YvanCywan** [comment]
Additionally, we could use milestones completion to trigger a release.

**Comment by magicaldave** [comment]
Granted, it IS possible to manually set the oldest compatible commits across each release, but that's an added support burden on our parts that I don't think makes a lot of sense in comparison to the more obvious and sane approach of just doing tagged releases in the first place.
